core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -340,4 +351,25 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
     sc.conf.getTimeAsSeconds("spark.mesos.rejectOfferDurationForUnmetConstraints", "120s")
   }
 
+  protected def getExecutorRelatedPort(sc: SparkContext, configProperty: String): Int = {
+    // either: spark.blockManager.port or spark.executor.port
+    sc.conf.getInt(s"$configProperty", 0)
+  }
+
+  protected def checkPorts(sc: SparkContext, ports: List[(Long, Long)]): Boolean = {
+
+    def checkIfInRange(port: Int, ps: List[(Long, Long)]): Boolean = {
+      ps.takeWhile(r => !(r._1 <= port && r._2 >= port)).size != ps.size
```

Comment:
This can be expressed in a simpler way using `exists`: `ps.exists(r => r._1 <= port & r._2 >= port)`